public/app/features/live/channel.test.ts

```diff
@@ -0,0 +1,24 @@
+import { LiveChannelAddress, LiveChannelEvent } from '@grafana/data';
+import { PartialObserver } from 'rxjs';
+import { CentrifugeLiveChannel } from './channel';
+
+describe('grafana live channel', () => {
+  it('accuratly counts the active observers', () => {
+    const chan = new CentrifugeLiveChannel('a/b/c', {} as LiveChannelAddress);
+    const noop: PartialObserver<LiveChannelEvent> = {
+      next: v => {},
+    };
+
+    expect(chan.getObserverCount()).toEqual(0);
+
+    const subA = chan.getStream().subscribe(noop);
+    expect(chan.getObserverCount()).toEqual(1);
+    const subB = chan.getStream().subscribe(noop);
+    expect(chan.getObserverCount()).toEqual(2);
+
+    subA.unsubscribe();
+    subB.unsubscribe();
+
+    expect(chan.getObserverCount()).toEqual(0);
```

Comment:
is there a way the channel can remove itself when the last subscriber unsubscribes? If you want to keep the channel open for some seconds after we could have a timeout that then checks if it still has no other subscribers?

Reply:
yes -- I'll look into that